src/api.rs

```diff
@@ -86,49 +93,49 @@ pub fn get_peer_count() -> Result<u32, String> {
         .map_err(|e| e.to_string())
 }
 
-pub fn scan_next_n_blocks(n: u32) -> Result<(), String> {
-    let sp_client = get_sp_client();
-
-    let sp_receiver = &sp_client.sp_receiver;
-    let scan_sk = sp_client.scan_sk;
-
+pub fn scan_next_n_blocks(path: String, label: String, n: u32) -> Result<(), String> {
     let electrum_client = create_electrum_client()
         .map_err(|e| e.to_string())?;
 
-    let scan_key_scalar: Scalar = scan_sk.into();
-
-    nakamotoclient::scan_blocks(n, sp_receiver, electrum_client, scan_key_scalar)
-        .map_err(|e| e.to_string())
+    match SpClient::try_init_from_disk(label, path) {
+        Ok(sp_client) => nakamotoclient::scan_blocks(n, sp_client, electrum_client).map_err(|e| e.to_string())?,
```

Comment:
i think we should also do `sp_client.save_to_disk()` here

Reply:
Good catch, I'd rather do it in scan_blocks itself though, since we take ownership of sp_client and don't return it.